conda-store-server/conda_store_server/server/app.py

```diff
@@ -343,13 +343,73 @@ async def favicon():
         return app
 
     def start(self):
+        import time
+        from threading import Thread
+
+        from conda_store_server import orm
+        from sqlalchemy.pool import QueuePool
+
         fastapi_app = self.init_fastapi_app()
 
         with self.conda_store.session_factory() as db:
             self.conda_store.ensure_settings(db)
             self.conda_store.ensure_namespace(db)
             self.conda_store.ensure_conda_channels(db)
 
+            # This ensures the database has no Worker table entires when the
+            # server starts, which is necessary for the worker to signal that
+            # it's ready via task_initialize_worker. Old Worker entries could
+            # still be in the database on startup after they were added on the
+            # previous run and the server was terminated.
+            #
+            # Note that this cleanup is deliberately done on startup because the
+            # server could be terminated due to a power failure, which would
+            # leave no chance for cleanup actions to run on shutdown.
+            #
+            # The database is used for worker-server communication because it
+            # will work regardless of celery_broker_url used, which can be Redis
+            # or just point to a database connection.
+            db.query(orm.Worker).delete()
+            db.commit()
+
+        def check_worker():
+            # Uses colors to make the output more visible
+            green = "\x1b[32m"
+            red = "\x1b[31m"
+            reset = "\x1b[0m"
+
+            # Creates a new DB connection since this will be run in a separate
+            # thread and connections cannot be shared between threads
+            session_factory = orm.new_session_factory(
+                url=self.conda_store.database_url,
+                poolclass=QueuePool,
+            )
+
+            # Waits in a loop for the worker to become ready, which is
+            # communicated by the worker via task_initialize_worker
+            while True:
+                with session_factory() as db:
+                    q = db.query(orm.Worker).first()
+                    if q is not None and q.initialized:
+                        self.log.info(f"{green}" "Worker initialized" f"{reset}")
+                        break
+
+                time.sleep(5)
```

Comment:
It would be nice to have this be passed in as a parameter with a default value so that for testing we could make the time shorter.